hack/build-go.sh

```diff
@@ -104,19 +121,19 @@ for arg in $args; do
             cd $arg
             go vet ./...
 
-            # always build and link the linux/amd64 binary
-            LINUX_NAME=${ARCH_BASENAME}-linux-amd64
+            # always build and link the binary based on CPU Architecture
+            LINUX_NAME=${ARCH_BASENAME}-linux-${ARCH}
```

Comment:
/cc @stu-gott

FYI, this will likely require some Dockerfile changes for build-go.sh consumers in other system (@zhlhahaha no Dockerfiles in this repo, so that you are not confused).

Reply:
@stu-gott just another ping so that you don't miss this.